File: app/src/main/java/k/bs/designpatternsp/behavioral/mediator/ja/ChatMediator.java
package k.bs.designpatternsp.behavioral.mediator.ja;

import android.os.Build;

import androidx.annotation.RequiresApi;

import java.util.ArrayList;
import java.util.List;

@RequiresApi(api = Build.VERSION_CODES.N)
public class ChatMediator {
    private List<ChatUser> users = new ArrayList<>();

    public void sendMessage(String msg, ChatUser user) {
        users.stream()
                .filter(u -> u != user)
                .forEach(u -> u.receive(msg));
    }

    public ChatMediator addUser(ChatUser user) {
        users.add(user);
        return this;
    }
}

File: app/src/main/java/k/bs/designpatternsp/pattern/mediator/kt/ChatMediator.kt
package k.bs.designpatternsp.pattern.mediator.kt

class ChatMediator {

    private val users: MutableList<ChatUser> = ArrayList()

    fun sendMessage(msg: String, user: ChatUser) {
        users
            .filter { it != user }
            .forEach { it.receive(msg) }
    }

    fun addUser(user: ChatUser): ChatMediator =
        apply { users.add(user) }
}
